File: src/main/java/com/violeth/blockjs/blockjs/jsinterface/FileReader.java
package com.violeth.blockjs.blockjs.jsinterface;

import java.io.File;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class FileReader {

    public static List<File> getListOfFiles(File folder) {
        List<File> files = new ArrayList<>();

        if (folder.listFiles() == null) {
            return files;
        }

        for (var fsEntry : Objects.requireNonNull(folder.listFiles())) {
            if (fsEntry.isFile()) {
                files.add(fsEntry.getAbsoluteFile());
            } else if (fsEntry.isDirectory()) {
                var subFiles = getListOfFiles(fsEntry);

                if (subFiles.size() > 0) {
                    files.addAll(subFiles);
                }
            }
        }

        return files;
    }
}

File: plugin/src/main/kotlin/io/github/blockjs/plugin/js/runtime/utils/FileReader.kt
package io.github.blockjs.plugin.js.runtime.utils

import java.io.File
import java.util.Objects

object FileReader {
    fun getListOfFiles(folder: File): MutableList<File> {
        val files = mutableListOf<File>()

        if (folder.listFiles() == null) {
            return files
        }

        for (fsEntry in Objects.requireNonNull(folder.listFiles())) {
            if (fsEntry.isFile()) {
                files.add(fsEntry.getAbsoluteFile())
            } else if (fsEntry.isDirectory()) {
                if (fsEntry.getName().equals("node_modules")) {
                    continue
                }

                val subFiles = getListOfFiles(fsEntry)

                if (subFiles.size > 0) {
                    files.addAll(subFiles)
                }
            }
        }
        return files
    }
}
